### ISSUE TITLE
JavaSymbolSolver unable to resolve an inner class defined in a base class

### ISSUE THREAD

**Issue by corradomio** [javaparser/javaparser#3100]
Given the following classes:

```
public class ConvolutionLayer {
    public static class Builder {}
}
```

```
public class Convolution2D extends ConvolutionLayer { }
```
```
public class Test {
    public static void main(String[] args) {
        ConvolutionLayer.Builder b = new Convolution2D.Builder();
    }
}
```

JavaSymbolResolver is not able to resolve "Convolution2D.Builder" because it try to find "Builder" inside "Convolution2D" and not also inside "ConvolutionLayer".

ref: Navigator.java, method: "findType()"

Note: I am analyzing Deeplearning4J.

**Comment by corradomio** [comment]
Another strange case.
Given the following classes:

```
public class Trie {
    public class Node { }
}
```
```
public class NodeTest {

    public void testNode() {
        Trie trie = new Trie();
        Trie.Node node = trie.new Node();
    }
}
```
JSS is not able to solve "Node" (as "Trie.Node") in " trie.new Node()"

**Comment by jlerbsc** [comment]
I cannot reproduce your issue with this test case 

    @Test
    void test() {
        String code = 
                "public class NodeTest {\n" + 
                "\n" + 
                "    public void testNode() {\n" + 
                "        Trie trie = new Trie();\n" + 
                "        Trie.Node node = trie.new Node();\n" + 
                "    }\n" + 
                "}";
        Path pathToSourceFile = adaptPath("src/test/resources/issue3100");
        TypeSolver typeSolver = new CombinedTypeSolver(
                new ReflectionTypeSolver(),
                new JavaParserTypeSolver(pathToSourceFile));
        ParserConfiguration config = new ParserConfiguration();
        StaticJavaParser.getConfiguration().setSymbolResolver(new JavaSymbolSolver(typeSolver));
        CompilationUnit cu = StaticJavaParser.parse(code);
        System.out.println(cu.toString());
        ObjectCreationExpr oce = cu.findAll(ObjectCreationExpr.class).get(1);
        System.out.println(oce.calculateResolvedType().describe());
    }

issue3100 directory contains

    public class Trie {
        public class Node { }
    }

**Comment by corradomio** [comment]
I use this type of code:

```
public static void main(String[] args) {
        String code =
            "public class NodeTest {\n" +
                "\n" +
                "    public void testNode() {\n" +
                "        Trie trie = new Trie();\n" +
                "        Trie.Node node = trie.new Node();\n" +
                "    }\n" +
                "}";
        Path pathToSourceFile = adaptPath("src/test/resources/issue3100");
        TypeSolver typeSolver = new CombinedTypeSolver(
            new ReflectionTypeSolver(),
            new JavaParserTypeSolver(pathToSourceFile));
        ParserConfiguration config = new ParserConfiguration();
        StaticJavaParser.getConfiguration().setSymbolResolver(new JavaSymbolSolver(typeSolver));
        CompilationUnit cu = StaticJavaParser.parse(code);
        System.out.println(cu.toString());

        (new VoidVisitorAdapter<Void>() {

            public void visit(ClassOrInterfaceType n, Void arg) {
                try {
                    n.resolve();
                }
                catch (Exception e) {
                    System.out.printf("%s %s\n", n, e);
                }
            }

        }).visit(cu, null);

    }
```

**Comment by corradomio** [comment]
Same for the previous bug (with Comvolution2D.java & ConvolutionLayer.java in "src/test/resources/issue3100"):

```
public class Issue3100a extends Issues {

    public static void main(String[] args) {
        String code =
            "public class Test {\n" +
                "\n" +
                "    public static void main(String[] args) {\n" +
                "        ConvolutionLayer.Builder b = new Convolution2D.Builder();\n" +
                "    }\n" +
                "}\n";
        Path pathToSourceFile = adaptPath("src/test/resources/issue3100");
        TypeSolver typeSolver = new CombinedTypeSolver(
            new ReflectionTypeSolver(),
            new JavaParserTypeSolver(pathToSourceFile));
        ParserConfiguration config = new ParserConfiguration();
        StaticJavaParser.getConfiguration().setSymbolResolver(new JavaSymbolSolver(typeSolver));
        CompilationUnit cu = StaticJavaParser.parse(code);
        System.out.println(cu.toString());

        (new VoidVisitorAdapter<Void>() {

            public void visit(ClassOrInterfaceType n, Void arg) {
                try {
                    n.resolve();
                }
                catch (Exception e) {
                    System.out.printf("%s %s\n", n, e);
                }
            }

        }).visit(cu, null);

    }
}
```

**Comment by jlerbsc** [comment]
> public class Trie {
>     public class Node { }
> }

This case seems to be resolved in the current version (3.25.10).
